mypy/types.py

```diff
@@ -284,6 +284,8 @@ class TypeOfAny(Enum):
     special_form = 'special_form'
     # Does this Any come from interaction with another Any?
     from_another_any = 'from_another_any'
+    # Does this Any come from an implementation limitation/bug?
+    implementation_artifact = 'implementation_artifact'
```

Comment:
Big +1 for adding this. I think I have seen other cases where this would be appropriate (but can't remember them now).